source/dofs/dof_tools_constraints.cc

```diff
@@ -1848,8 +1848,8 @@ namespace DoFTools
 
           const unsigned int dofs_per_face =
             face_1->get_fe(face_1->nth_active_fe_index(0)).dofs_per_face;
-          FullMatrix<double> child_transformation(dofs_per_face, dofs_per_face);
-          FullMatrix<double> subface_interpolation(dofs_per_face,
+          FullMatrix<number> child_transformation(dofs_per_face, dofs_per_face);
+          FullMatrix<number> subface_interpolation(dofs_per_face,
```

Comment:
Can't you leave this as `FullMatrix<double>` and drop your modifications to `get_subface_interpolation_matrix`?

Reply:
That is right!  `subface_interpolation` can be `FullMatrix<double> ` and `child_transformation` can be `FullMatrix<number>`. I had to add a few instantiations for `FullMatrix`, notably `FullMatrix<S1>::mmult<S2>`.